test/test_primal_dual_objective.py

```diff
@@ -71,11 +76,14 @@ def test_dual_obj():
     disp = init_gap + surface.heights() + offset
     init_pressure = substrate.evaluate_force(disp)
 
-    res = optim.minimize(system.dual_objective(offset, gradient=True),
-                         init_pressure,
-                         method='L-BFGS-B', jac=True,
-                         bounds=bnds,
-                         options=dict(gtol=1e-8, ftol=1e-20))
+    # res = optim.minimize(system.dual_objective(offset, gradient=True),
+    #                      init_pressure,
+    #                      method='L-BFGS-B', jac=True,
+    #                      bounds=bnds,
+    #                      options=dict(gtol=1e-8 * system.area_per_pt,
+    #                                   ftol=1e-20))
+    res = system.dual_minimize_proxy(x0=disp, offset=offset, gtol=1e-8)
```

Comment:
We did a change in the code. 
Name: 

x0 -> initial_displacements

Please always allow to give the actual variable as initial guess. 

This means for the dual objective: 

initial_forces

for the primal 

initial_gaps.

Feel free to add initial_forces bzw. initial_gaps or even initial_displacements for convenience.